src/index.js

```diff
@@ -11,161 +13,25 @@ export default function json(options = {}) {
-							start: start + 7, // 'export '.length
-							end,
-							declarations: [
-								{
-									type: 'VariableDeclarator',
-									start: start + 7 + declarationType.length + 1, // `export ${declarationType} `.length
-									end: end - 1,
-									id: {
-										type: 'Identifier',
-										start: start + 7 + declarationType.length + 1, // `export ${declarationType} `.length
-										end: start + 7 + declarationType.length + 1 + key.length, // `export ${declarationType} ${key}`.length
-										name: key
-									},
-									init: {
-										type: 'Literal',
-										start: start +
-											7 +
-											declarationType.length +
-											1 +
-											key.length +
-											3, // `export ${declarationType} ${key} = `.length
-										end: end - 1,
-										value: null,
-										raw: 'null'
-									}
-								}
-							],
-							kind: declarationType
-						},
-						specifiers: [],
-						source: null
-					});
-
-					char = end + 1;
-					code += `${declaration}\n`;
-				});
-
-				const defaultExportNode = {
-					type: 'ExportDefaultDeclaration',
-					start: char,
-					end: null,
-					declaration: {
-						type: 'ObjectExpression',
-						start: char + 15,
-						end: null,
-						properties: []
-					}
-				};
-
-				char += 17 + indent.length; // 'export default {\n\t'.length'
-
-				const defaultExportRows = validKeys
-					.map(key => {
-						const row = `${key}: ${key}`;
-
-						const start = char;
-						const end = start + row.length;
-
-						defaultExportNode.declaration.properties.push({
-							type: 'Property',
-							start,
-							end,
-							method: false,
-							shorthand: false,
-							computed: false,
-							key: {
-								type: 'Identifier',
-								start,
-								end: start + key.length,
-								name: key
-							},
-							value: {
-								type: 'Identifier',
-								start: start + key.length + 2,
-								end,
-								name: key
-							},
-							kind: 'init'
-						});
-
-						char += row.length + (2 + indent.length); // ',\n\t'.length
-
-						return row;
-					})
-					.concat(
-						invalidKeys.map(key => `"${key}": ${JSON.stringify(data[key])}`)
-					);
-
-				code += `export default {\n${indent}${defaultExportRows.join(`,\n${indent}`)}\n};`;
-				ast.body.push(defaultExportNode);
-
-				const end = code.length;
-
-				defaultExportNode.declaration.end = end - 1;
-				defaultExportNode.end = end;
+				return {code: `export default ${json};\n`, map: {mappings: ''}};
 			}
 
-			ast.end = code.length;
-
-			return { ast, code, map: { mappings: '' } };
+			let namedExportCode = '';
+			const defaultExportRows = [];
+			Object.keys(data).forEach(key => {
+				if (key === makeLegalIdentifier(key)) {
+					defaultExportRows.push(`${key}: ${key}`);
+					namedExportCode += `export ${declarationType} ${key} = ${JSON.stringify(data[key])};\n`;
+				} else {
+					defaultExportRows.push(`"${key}": ${JSON.stringify(data[key])}`);
```

Comment:
This is a lot nicer than what I did in https://github.com/rollup/rollup-pluginutils/pull/29/files actually.